server/routes/viewModels/taskLists/caTasks.js

```diff
@@ -38,23 +38,17 @@ module.exports = {
     const optOutUnstarted = optOut === 'UNSTARTED'
     const optOutRefused = optOut === 'DONE' && !optedOut
 
-    return [
-      eligibilityTask({ visible: true }),
-      informOffenderTask({ visible: eligibilityDone && optOutUnstarted && !optedOut }),
-      curfewAddress({ decisions, tasks, visible: eligible }),
-      riskManagement.edit({ decisions, tasks, visible: addressUnsuitable }),
-      caSubmitToDm.refusal({ decisions, visible: allowedTransition === 'caToDmRefusal' }),
-      caSubmitBassReview({
-        decisions,
-        tasks,
-        visible: optOutRefused && bassReferralNeeded && allowedTransition !== 'caToDmRefusal',
-      }),
-      caSubmitAddressReview({
-        decisions,
-        tasks,
-        visible: optOutRefused && !bassReferralNeeded && allowedTransition !== 'caToDmRefusal',
-      }),
-    ].filter((task) => task.visible)
+    const context = { decisions, tasks, allowedTransition }
+
+    return tasklist(context, [
+      [eligibilityTask, true],
+      [informOffenderTask, eligibilityDone && optOutUnstarted && !optedOut],
```

Comment:
I went for the most terse representation: `[taskFunction, visible]`, could go less terse more expressive.

Also could move to `visible = true` by default